services/src/datasets/external/gfbio.rs

```diff
@@ -0,0 +1,661 @@
+                    ("/DataSets/DataSet/Metadata/IPRStatements/Licenses/License/Text".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/Gathering/SiteCoordinateSets/SiteCoordinates/CoordinatesLatLong/SpatialDatum".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/MultiMediaObjects/MultiMediaObject/FileURI".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/ContentContacts/ContentContact/Name".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/Gathering/Country/Name".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/SourceInstitutionID".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/MultiMediaObjects/MultiMediaObject/IPR/Licenses/License/URI".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/MultiMediaObjects/MultiMediaObject/Format".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/RecordURI".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Metadata/IPRStatements/Licenses/License/Details".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/Identifications/Identification/Result/TaxonIdentified/HigherTaxa/HigherTaxon/HigherTaxonName".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/RecordBasis".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/DateLastEdited".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/Gathering/Country/ISO3166Code".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/MultiMediaObjects/MultiMediaObject/IPR/Licenses/License/Details".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/TechnicalContacts/TechnicalContact/Name".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Metadata/Description/Representation/URI".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/TechnicalContacts/TechnicalContact/Email".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/UnitID".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Metadata/Description/Representation/Title".to_owned(), FeatureDataType::Text),
+                    ("/DataSets/DataSet/Units/Unit/Gathering/LocalityText".to_owned(), FeatureDataType::Text)]
+                    .iter()
+                    .cloned()
+                    .collect()
+            };
+
+            let result_descriptor = meta.result_descriptor().await?;
+
+            if result_descriptor != expected {
+                eprintln!("{:?} != {:?}", result_descriptor, expected);
+                return Err(Error::TestFail);
+            }
```

Comment:
It looks odd to have an error just for testing purposes.
Moreover, you could just throw a panic.

If you want to ensure that you still drop the schema, I guess there are two options:
1. Use [catch_unwind](https://doc.rust-lang.org/std/panic/fn.catch_unwind.html) to get any panic and then do the cleanup.
2. Set a panic hook (https://doc.rust-lang.org/std/panic/fn.set_hook.html) and perform the cleanup.

What do you think?

Reply:
I agree that it's not pretty. I tried both, but they work only with sync code but not async. I also tried https://docs.rs/futures/0.3.5/futures/future/trait.FutureExt.html#method.catch_unwind but that one doesn't work with postgres stuff because of thread boundaries or something.

Reply:
Okay, but then you could say
`async fn test(db_config: &config::Postgres, test_schema: &str) -> Result<(), String> {`
and then 
```
    if result_descriptor != expected {
        return Err(format!("{:?} != {:?}", result_descriptor, expected));
    }
```
and we wouldn't need TestFail in our list of errors of the services crate.

Reply:
e197421ffb6387bfb51c07656dc0f12bb02ea79b